### ISSUE TITLE
[2.9.1-rc1] Missing 'use strict' issue introduced

### ISSUE THREAD

**Issue by hughes** [jshint/jshint#2759]
I've noticed several files with "use strict" in the top-level function have started failing jshint since 2.9.1-rc1 was published, with the message of a missing "use strict" statement. Anyone else noticed this?

**Comment by nicolo-ribaudo** [comment]
Can you provide an example?

**Comment by benatwork99** [comment]
Yes I have noticed this too.

See: https://travis-ci.org/smart-facility/cognicity-reports-powertrack/builds/90874359

**Comment by nicolo-ribaudo** [comment]
@benatwork99 You get different warnings:

> Incompatible values for the 'strict' and 'globalstrict' linting options.

JSHint reports this warning when you use both `strict: global` and `globalstrict` (#2703)

> Read only.

It looks like `-W020` doesn't work anymore. Can you open a new issue?

**Comment by randspy** [comment]
I also have this problem.

**Comment by nicolo-ribaudo** [comment]
@randspy Can you provide an example / a link to a file to reproduce this issue?

**Comment by hughes** [comment]
Might have actually just been a bugfix related to `strict: "global"` that revealed something in my project that was actually not right. The failing files all looked like this:

```
/*globals describe*/
describe('foo', function() {
  'use strict';
  // ...
});
```

In 2.8, this was passing with `"strict": "global"` in my .jshintrc

**Comment by nicolo-ribaudo** [comment]
The strict option requires statements different from function declarations and IIFEs to be in strict mode. `strict: true` doesn't warn about global level code (while `strict: global` does) because it doesn't allow the use of a global "use strict" directive.

In 2.8.0, using `strict: true, globalstrict: true` you didn't get the error because jshint didn't warn when invocations and other expressions were not in strict mode: (fixed in #2678)
```js
describe(); // jshint didn't warn
var a = 2;   // 'Missing "use strict" statement.', ok
```

> In 2.8, this was passing with `"strict": "global"` in my .jshintrc

`strict: global` was introduced in JSHint 2.9.0, so in 2.8.0 it was interpreted as `strict: true`

**Comment by benatwork99** [comment]
@nicolo-ribaudo You are correct re. the different warnings, I should have read more carefully. Thanks for opening the new issue, I only got back to this today.

**Comment by jugglinmike** [comment]
I think this is safe to close, but @lukeapage's recent assignment of the "Regression" label suggests otherwise. Luke: can you explain the regression here?

**Comment by jugglinmike** [comment]
Although the distinction between "regression" and "bug fix" is sometimes
subjective, we're lucky in this case because it's fairly cut-and-dry. JSHint
2.8.0's documentation for the `strict` option explicitly described the
sub-optimal behavior:

> *Note:* This option enables strict mode for function scope only. It
> *prohibits* the global scoped strict mode because it might break
> third-party widgets on your page. If you really want to use global
> strict mode, see the *globalstrict* option.

...so changing that is a regression.

It's taken me a while to recognize that because the change is otherwise an
improvement (it helps users write more code in strict mode in a safe way). It
wasn't until gh-2810 that I reviewed the old documentation.

@hughes I appreciate your courtesy when you say, "it probably should have been
failing all along," but I have a feeling many of our users would not be so
forgiving if the new error you reported appeared in a minor release.

For anyone following this discussion: I've submitted gh-2822 to preserve the
legacy behavior. My short-term recommendation would be to use `strict: global`
for Node.js code. We can make this unnecessary in JSHint 3; `strict: true` can
"do the right thing" depending on the rest of the configuration.
